pkg/cloudprovider/aws/providers/providers.go

```diff
@@ -0,0 +1,97 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package providers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"k8s.io/client-go/kubernetes"
+	"knative.dev/pkg/logging"
+
+	"github.com/aws/karpenter/pkg/utils/project"
+)
+
+const (
+	// CacheTTL restricts QPS to AWS APIs to this interval for verifying setup
+	// resources. This value represents the maximum eventual consistency between
+	// AWS actual state and the controller's ability to provision those
+	// resources. Cache hits enable faster provisioning and reduced API load on
+	// AWS APIs, which can have a serious impact on performance and scalability.
+	// DO NOT CHANGE THIS VALUE WITHOUT DUE CONSIDERATION
+	CacheTTL = 60 * time.Second
+	// CacheCleanupInterval triggers cache cleanup (lazy eviction) at this interval.
+	CacheCleanupInterval = 10 * time.Minute
+)
+
+type Providers struct {
```

Comment:
I'm not sure I'm sold on this abstraction. Is this used in multiple places? If not, I'd just keep it as part of cloudprovider.go